src/datahike/tools.cljc

```diff
@@ -21,3 +23,10 @@
 (defn get-time []
   #?(:clj (java.util.Date.)
      :cljs (js/Date.)))
+
+(defmacro raise [& fragments]
```

Comment:
We could enforce here in the macro that the first argument to `raise` must be a message string and the second argument must be a map containing a `:type` key.

Reply:
I think we should also have a `:level` set by default, which is probably always going to be `:error` (?)

Reply:
Yeah, sometimes we don't need all info level notifications.